logstage/logstage-sink-file/src/main/scala/com/github/pshirshov/izumi/models/Rotation.scala

```diff
@@ -0,0 +1,40 @@
+package com.github.pshirshov.izumi.models
+
+import scala.util.{Success, Try}
+
+trait Rotation {
+  def rotateFileId(fileId: FileId, state: SinkState): Try[(FileId, SinkState)]
+
+  def filesLimit: Option[Int]
+}
+
+object Rotation {
+
+  case object DisabledRotation extends Rotation {
+    override def rotateFileId(fileId: FileId, state: SinkState): Try[(FileId, SinkState)] = {
+      Success((fileId, state))
+    }
+
+    override lazy val filesLimit: Option[FileId] = None
```

Comment:
`lazy val` doesn't do anything here. Consider not using lazy val unless it's required, as it has a performance penalty https://stackoverflow.com/questions/3041253/whats-the-hidden-cost-of-scalas-lazy-val